### ISSUE TITLE
Remove gem awesome_nested_set

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#11636]
### What we should change and why (this is tech debt)

It's providing a tree structure for Spree::Taxon but we don't use the tree structure. I think that the UI is actually not saving it when you try. If we ever wanted a tree structure in the future, it would be better to use [ancestry](https://github.com/stefankroes/ancestry) for it's simpler design and better update performance.

There are several related issues that may be resolved at the same time:

* #9677
* #9386
* #6033

### Context

* #11631

### Impact and timeline

No significant impact at the moment. But simplifying code always reduces maintenance and performance.

**Comment by HillaryOkello** [comment]
Can I work on this?

**Comment by sigmundpetersen** [comment]
Sure thanks @HillaryOkello , go ahead 🙂

**Comment by HillaryOkello** [comment]
I created a PR for this issue but I'm having an issue with the tests, I'm working on fixing them https://github.com/openfoodfoundation/openfoodnetwork/pull/11808

**Comment by sigmundpetersen** [comment]
Yes, thank you @HillaryOkello 
The PR is waiting on code review, so you will get feedback from a dev as soon as they get to it 👍

**Comment by HillaryOkello** [comment]
Thank you for the response @sigmundpetersen 
I will keep trying to fix them as I wait for a code review 😄 👍

**Comment by HillaryOkello** [comment]
Hello @sigmundpetersen,

Sorry, I've taken too long on this issue. I took a break and I'm back now.
However, I find this issue to be more complicated because many changes are required and removing some pieces of code is affecting some areas I don't understand quite well yet.

Is it okay if I pick another issue to work on?

**Comment by sigmundpetersen** [comment]
Hey @HillaryOkello 
Sure, no worries! Just pick another one 👍

**Comment by sigmundpetersen** [comment]
Thank you for your efforts on this one and thanks for the feedback!

**Comment by thebigw4lrus** [comment]
hey @sigmundpetersen I want to take over this one to warm up. WDYTH?

**Comment by sigmundpetersen** [comment]
Sure thanks @thebigw4lrus , go ahead! 🎉 Thank you
Don't hesitate to ask if there's anything!

**Comment by thebigw4lrus** [comment]
hey @sigmundpetersen , as I started digging into this issue, I got a few questions:

1. I see there is a taxonomy model, related 1-n with taxons models. I see in the admin interface once I create a taxonomy, two entities will be created: Taxonomy and Taxons. If we try to remove the tree representation of taxonomies, it looks to my that taxon purpose was only to model this. Should we remove taxon model and stick only with taxonomy?. 

2. Looks like the Frontend needs to change as well: Now the Frontend tries to handle the trees unsuccessfully (it does not create them). is this part of the scope of the ticket?.

The way I see it there are two ways to move forward:

1. Remove the Gem, Clean the database structure for taxon model, and clean every single call to "parent" or "children" and assume a flat model.

2. The above (remove the gem), including also the FE changes (It would turn into a simple CRUD of taxonomy) and removing Taxon model. This way hypothetical Admin Simple Taxonomy CRUD would match with the database structure.

I am asking this to reassure on the ticket scope, as if we go for the path number 2, It would imply changes in several controllers outside of the taxon controller itself.

Thanks!

**Comment by sigmundpetersen** [comment]
Thanks @thebigw4lrus .
Let's ask the @openfoodfoundation/core-devs if they have any feedback on this 👍

**Comment by mkllnk** [comment]
Very good analysis, thank you!

> Should we remove taxon model and stick only with taxonomy?

It's the other way round. Taxons are referenced by products and they contain important data. The taxonomy is a group of taxons and not used anywhere. It can be removed, I believe.

All the taxon fields relating to the tree structure can also be removed. We may need to keep the position for ordering, I'm not sure.

> Looks like the Frontend needs to change as well

Yes, this is part of the scope. It can be a complete rewrite and it can be super simple like a standard Rails view to edit the taxon model (index, create, update). Possibly with positioning, not sure.

Does this give you a clear path forward?

**Comment by thebigw4lrus** [comment]
It does @mkllnk , starting my engines then :100:

**Comment by sigmundpetersen** [comment]
Hi @thebigw4lrus hope everything is fine 🙂 
Do you still want to work on this?

**Comment by wandji20** [comment]
I would love to work on this one.

**Comment by mkllnk** [comment]
Great, I assigned you, @wandji20.
